build.gradle

```diff
@@ -235,6 +235,38 @@ project(':iceberg-data') {
   }
 }
 
+project(':iceberg-flink') {
+  apply plugin: 'scala'
+
+  dependencies {
+    compile project(':iceberg-api')
+    compile project(':iceberg-common')
+    compile project(':iceberg-core')
+    compile project(':iceberg-data')
+    compile project(':iceberg-orc')
+    compile project(':iceberg-parquet')
+    compile project(':iceberg-arrow')
+    compile "org.apache.flink:flink-streaming-java_2.11::tests"
```

Comment:
Does Flink have a Scala 2.12 version? Will it cause a problem to rely on 2.11?

Reply:
This is compiling the Iceberg sink against the Flink Scala 2.11 API, right? But Scala isn't binary compatible between minor releases, so Jars compiled for 2.11 may not work with 2.12. How should we avoid problems with binary compatibility?

Reply:
IMO,  we may need to load the corresponding  flink Jars  (which is depending on scala ) for the configured scala version.  For example,  if we choose to use the scala 2.12, then the iceberg-flink module should choose to use all the flink-xxx_2.12 jars (should throw error if the flink-xx_2.12 does not exist).

This may need to write few gradlew scripts or methods.  I think it could be considered in a separate issue. There're also some other modules such as `iceberg-spark` which should handle this issue too.

Reply:
My main concern is that if we compile against 2.11, it won't work in 2.12. Then we would need two separate builds.

Reply:
I think so, building two `iceberg-flink` jars and deploying them to the repo, so that the downstream users can choose to use the correct `iceberg-flink` jar based on their scala version.